src/textual/layouts/horizontal.py

```diff
@@ -23,10 +23,23 @@ def arrange(
         x = max_height = Fraction(0)
         parent_size = parent.outer_size
 
+        child_styles = [child.styles for child in children]
+        box_margins = [styles.margin for styles in child_styles]
```

Comment:
Is there any reason you didn't do
`box_margins = [child.styles.margin for child in children]`?

Reply:
Yes, an optimization I forgot to port from horizontal.